### PR TITLE
pkgsCross.x86_64-unknown-redox: fix evaluation of stdenv

### PR DESCRIPTION
**Pull request by Mic92**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by aaronjanse — APPROVED** [review]
I've confirmed that this builds on Linux, and this LGTM. Thank you @Mic92!

File: pkgs/development/libraries/relibc/default.nix [hunk 1]
```diff
@@ -73,6 +73,6 @@ redoxRustPlatform.buildRustPackage rec {
     description = "C Library in Rust for Redox and Linux";
     license = licenses.mit;
     maintainers = [ maintainers.aaronjanse ];
-    platforms = platforms.redox;
+    platforms = platforms.redox ++ [ "x86_64-linux" ];
```
Comment (r-burns) [inline pkgs/development/libraries/relibc/default.nix:76]: Upstream claims to support aarch64-linux too

Comment (Mic92) [reply pkgs/development/libraries/relibc/default.nix:76]: I tried but there is some issue with the vendoring on aarch64.

Comment (r-burns) [reply pkgs/development/libraries/relibc/default.nix:76]: Ah ok, thanks for giving it a try.

**Comment by r-burns** [comment]
Change is good, but do you know why this didn't eval before? We shouldn't need linux in the supported platforms to be able to eval/build pkgsCross.x86_64-unknown-redox.stdenv.cc.libc, as far as I'm aware

**Comment by aaronjanse** [comment]
The provided "seed" binaries are compiled for Linux, which we then use to compile (on Linux) relibc for whichever host/target platform we want

**Comment by r-burns** [comment]
Yes, but shouldn't it still eval? To give another example, netbsd.libc isn't supported on linux either, but I'm still able to eval+build pkgsCross.x86_64-netbsd.netbsd.libc. Shouldn't this have also been the case for relibc?

**Comment by Mic92** [comment]
> Yes, but shouldn't it still eval? To give another example, netbsd.libc isn't supported on linux either, but I'm still able to eval+build pkgsCross.x86_64-netbsd.netbsd.libc. Shouldn't this have also been the case for relibc?

I agree, I suppose `buildPackages.relibc` is referenced somewhere where it should not be.

**Pull request merged**